sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+    f"""Calculate the thermal X-ray emission (lines + continuum) from the solar atmosphere.
+
+    The emission is calculated as a function of temperature and emission measure.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
+    determined by the file from which the comtinuum parameters are loaded.
+    To change the file used, see the setup_continuum_parameters() function.
+    
+    {doc_string_params}"""
+    energy_edges_keV = energy_edges.to_value(u.keV)
+    temperature_K = temperature.to_value(u.K)
+    if temperature.isscalar:
+        temperature_K = np.array([temperature_K])
+    continuum_flux = _continuum_emission(energy_edges_keV, temperature_K, abundance_type,
+                                         relative_abundances)
+    line_flux = _line_emission(energy_edges_keV, temperature_K, abundance_type, relative_abundances)
+    flux = ((continuum_flux + line_flux) * emission_measure / 
+                (4 * np.pi * observer_distance**2))
+    if temperature.isscalar and emission_measure.isscalar:
+        flux = flux[0]
+    return flux
+
+
+@u.quantity_input(energy_edges=u.keV,
+                  temperature=u.K,
+                  emission_measure=(u.cm**(-3), u.cm**(-5)),
+                  observer_distance=u.cm)
+def continuum_emission(energy_edges,
+              temperature,
+              emission_measure,
+              abundance_type="sun_coronal",
+              relative_abundances=None,
+              observer_distance=(1*u.AU).to(u.cm)):
+    f"""Calculate the thermal X-ray continuum emission from the solar atmosphere.
+
+    The emission is calculated as a function of temperature and emission measure.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
+    determined by the file from which the comtinuum parameters are loaded.
+    To change the file used, see the setup_continuum_parameters() function.
+
+    {doc_string_params}"""
+    temperature_K = temperature.to_value(u.K)
+    if temperature.isscalar:
+        temperature_K = np.array([temperature_K])
+    flux = _continuum_emission(energy_edges.to_value(u.keV), temperature_K,
+                               abundance_type=abundance_type,
+                               relative_abundances=relative_abundances)
+    flux *= emission_measure / (4 * np.pi * observer_distance**2)
+    if temperature.isscalar:
+        flux = flux[0]
+    return flux
+
+def _continuum_emission(energy_edges_keV,
+                        temperature_K,
+                        abundance_type,
+                        relative_abundances):
+    """
+    Calculates emission-measure-normalized X-ray continuum emission at the source.
+
+    Output must be multiplied by emission measure and divided by 4*pi*observer_distance**2
+    to get physical values.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
```

Comment:
As of now, all are included.  But the way this is designed, that information belongs in the `setup_continuum_parameters` docstring.  Perhaps this docstring should direct the user there and the default made more clearly there.  It would be easier for the user if we just said it here, but then that's more places we need to edit if we ever change the default, and hence more scope for misleading documentation.  It's a tricky trade-off.  What do you think?